intake_esm/_search.py

```diff
@@ -4,6 +4,8 @@
 import numpy as np
 import pandas as pd
 
+from .utils import unpack_iterable_column
```

Comment:
Since this function is only used in `_search.py`, how about we move this function from `utils.py` to `_search.py` module?